netebpfext/net_ebpf_ext_sock_addr.c

```diff
@@ -77,6 +85,100 @@ _Guarded_by_(_net_ebpf_ext_sock_addr_lock) static LIST_ENTRY _net_ebpf_ext_redir
 _Guarded_by_(_net_ebpf_ext_sock_addr_lock) static LIST_ENTRY _net_ebpf_ext_connect_context_list;
 static uint32_t _net_ebpf_ext_connect_context_count = 0;
 
+static SECURITY_DESCRIPTOR* _net_ebpf_ext_security_descriptor_admin = NULL;
+static ACL* _net_ebpf_ext_dacl_admin = NULL;
+
+//
+// sock_addr helper functions.
+//
+static uint32_t
+_get_process_id()
+{
+    return (uint32_t)(uintptr_t)PsGetCurrentProcessId();
+}
+
+static uint32_t
+_get_thread_id()
+{
+    return (uint32_t)(uintptr_t)PsGetCurrentThreadId();
+}
+
+static uint64_t
+_ebpf_sock_addr_get_current_pid_tgid()
+{
+    net_ebpf_sock_addr_t* sock_addr_ctx = NULL;
+    ASSERT(_net_ebpf_sock_addr_get_program_context != NULL);
+    _net_ebpf_sock_addr_get_program_context((void**)&sock_addr_ctx);
+    ASSERT(sock_addr_ctx != NULL);
+
+    if (sock_addr_ctx == NULL) {
+        return (((uint64_t)_get_process_id() << 32) | _get_thread_id());
+    }
+
+    return (sock_addr_ctx->process_id << 32 | _get_thread_id());
+}
+
+static uint64_t
+_ebpf_sock_addr_get_current_logon_id(_In_ const bpf_sock_addr_t* ctx)
+{
+    uint64_t logon_id = 0;
+    net_ebpf_sock_addr_t* sock_addr_ctx = CONTAINING_RECORD(ctx, net_ebpf_sock_addr_t, base);
+    logon_id = *(uint64_t*)(&(sock_addr_ctx->access_information->AuthenticationId));
```

Comment:
no assert here unlike the previous function? And if there are any code path where this can be NULL, we should handle this instead of assert?

Reply:
Here the `net_ebpf_sock_addr_t` is retrieved from the context itself. In the previous case, it is calling `ebpfcore` API to fetch the context.